drivers/digital-io/max149x6/iio_max14916.c

```diff
@@ -0,0 +1,256 @@
+/***************************************************************************//**
```

Comment:
Please test this

Reply:
i will look again at the code for the iio driver, i noticed other bugs myself.

Reply:
I tested it, and a bool array ch_enable created by the user and transmitted through the iio_desc_init_param tells the iio_init function where the iio_setup_channels is called which channels will be active, and works when using iio oscilloscope to see the active channels.